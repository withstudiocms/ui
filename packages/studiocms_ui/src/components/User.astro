---
import { Image } from 'astro:assets';
import Icon from '../utils/Icon.astro';

interface Props {
	name: string;
	description: string;
	avatar?: string;
	class?: string;
  loading?: 'eager' | 'lazy'; 
};

const { name, description, avatar, class: className, loading } = Astro.props;
---
<div class="sui-user-container" class:list={[ className ]}>
  <div class="sui-avatar-container">
    {avatar ? (
      <Image src={avatar} inferSize loading={loading} alt={name} class="sui-avatar-img" />
    ) : (
      <Icon name='user' width={24} height={24} />
    )}
  </div>
  <div class="sui-text-content">
    <span class="sui-name">{name}</span>
    <span class="sui-description">{description}</span>
  </div>
</div>
<style>
  .sui-user-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 1rem;
  }

  .sui-avatar-container {
    width: 2.5rem;
    height: 2.5rem;
    background-color: hsl(var(--default-base));
    border-radius: 3rem;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    border: 1px solid hsl(var(--border));
  }

  .sui-avatar-img {
    width: 100%;
    height: auto;
  }

  .sui-text-content {
    display: flex;
    flex-direction: column;
    gap: .125rem;
  }

  .sui-name {
    font-size: 1em;
    font-weight: 600;
  }

  .sui-description {
    font-size: .875em;
    font-weight: 400;
    color: hsl(var(--text-muted));
  }
</style>
