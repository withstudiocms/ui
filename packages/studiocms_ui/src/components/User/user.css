.sui-user-container {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1rem;
}

.sui-avatar-container {
  width: 2.5rem;
  height: 2.5rem;
  background-color: hsl(var(--default-base));
  border-radius: var(--radius-full);
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  border: 1px solid hsl(var(--border));
}

.sui-avatar-img {
  width: 100%;
  height: auto;
}

.sui-text-content {
  display: flex;
  flex-direction: column;
  gap: .125rem;
}

.sui-name {
  font-size: 1em;
  font-weight: 600;
}

.sui-description {
  font-size: .875em;
  font-weight: 400;
  color: hsl(var(--text-muted));
}
