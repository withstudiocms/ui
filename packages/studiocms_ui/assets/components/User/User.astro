---
// @ts-ignore
import { Image } from 'astro:assets';
import Icon from '../Icon/Icon.astro';
import './user.css';

/**
 * The props for the User component.
 */
interface Props {
	/**
	 * The name of the user.
	 */
	name: string;
	/**
	 * The description of the user. Could be a role, a handle, etc.
	 */
	description: string;
	/**
	 * The avatar of the user. Either a URL to an image or an imported image.
	 */
	avatar?: string;
	/**
	 * Additional classes to apply to the user container.
	 */
	class?: string;
	/**
	 * The loading strategy for the image. Defaults to `lazy`.
	 */
	loading?: 'eager' | 'lazy';
}

const { name, description, avatar, class: className, loading = 'lazy' } = Astro.props;
---
<div class="sui-user-container" class:list={[ className ]}>
  <div class="sui-avatar-container">
    {avatar ? (
      <Image src={avatar} inferSize loading={loading} alt={name} class="sui-avatar-img" />
    ) : (
      <Icon name='user' width={24} height={24} role='img' aria-label={`Placeholder avatar for ${name}`} />
    )}
  </div>
  <div class="sui-text-content">
    <span class="sui-name">{name}</span>
    <span class="sui-description">{description}</span>
  </div>
</div>
