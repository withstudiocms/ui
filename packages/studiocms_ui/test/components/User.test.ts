import { describe, expect } from 'vitest';
import User from '../../src/components/User/User.astro';
import { test } from '../fixtures/vitest/AstroContainer';

describe('User Component', () => {
	test('renders User component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(User, 'User', {
			props: {
				name: 'John Doe',
				description: 'Administrator',
			},
		});
		expect(result).toMatchSnapshot();
	});

	test('renders User component with image correctly', async ({ renderComponent }) => {
		const result = await renderComponent(User, 'User', {
			props: {
				name: 'Jane Smith',
				description: 'Editor',
				avatar: 'https://seccdn.libravatar.org/static/img/mm/80.png',
			},
		});
		expect(result).toMatchSnapshot();
	});
});
