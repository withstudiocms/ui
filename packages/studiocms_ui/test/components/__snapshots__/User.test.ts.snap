// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Component > renders User component correctly 1`] = `"<div class="sui-user-container"> <div class="sui-avatar-container"> <svg style="min-width: 24px" xmlns="http://www.w3.org/2000/svg" prefix="heroicons" role="img" aria-label="Placeholder avatar for John Doe" height="24" width="24" viewbox="0 0 24 24"><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M15.75 6a3.75 3.75 0 1 1-7.5 0a3.75 3.75 0 0 1 7.5 0M4.501 20.118a7.5 7.5 0 0 1 14.998 0A17.9 17.9 0 0 1 12 21.75c-2.676 0-5.216-.584-7.499-1.632"/></svg> </div> <div class="sui-text-content"> <span class="sui-name">John Doe</span> <span class="sui-description">Administrator</span> </div> </div>"`;

exports[`User Component > renders User component with image correctly 1`] = `"<div class="sui-user-container"> <div class="sui-avatar-container"> <img src="https://seccdn.libravatar.org/static/img/mm/80.png" data-image-component="true" loading="lazy" alt="Jane Smith" decoding="async" fetchpriority="auto" width="80" height="80" class="sui-avatar-img"> </div> <div class="sui-text-content"> <span class="sui-name">Jane Smith</span> <span class="sui-description">Editor</span> </div> </div>"`;
